mountpoint-s3/src/prefetch.rs

```diff
@@ -155,6 +158,15 @@ where
             "read"
         );
 
+        // Currently, we set preferred part size to the current read size.
+        // Our assumption is that the read size will be the same for most sequential
+        // read and it can be aligned to the size of prefetched chunks.
+        //
+        // We should also put some bounds on this value in case the read size
+        // is too small.
+        let min_part_size = 128 * 1024;
+        self.preferred_part_size = length.max(min_part_size);
```

Comment:
For choosing 128k, it's the linux readahead size and seem to be a reasonable minimum value. I will update the comment, also happy to change if you have better suggestion.

I'm not sure we should put a maximum value though. If the read size is really big then we will have to combine data from multiple parts and the `extend` operation is quite [expensive](https://github.com/awslabs/mountpoint-s3/blob/cc93e4283b2212472bc4e4b9abc09cf72d7eaab3/mountpoint-s3/src/prefetch/checksummed_bytes.rs#L74-L92).

Reply:
but if we change the logic the use max value between last `preferred_part_size` and  current `length` I think there should be a maximum, otherwise it will keep growing bigger.

Reply:
The effective maximum is the client's part size anyway, so probably we should enforce it here just to be explicit.